Cleaned RA yaml rules

reaperrr opened the pull request:
Sanitizes defaults and removes some duplication.

reaperrr committed:
Cleaned RA yaml rules.

Sanitized defaults.

abcdefg30 commented on mods/ra/rules/defaults.yaml:
```diff
@@ -577,31 +552,30 @@
+	Tooltip:
+		GenericName: Destroyed Helicopter
+	FallsToEarth:
+		Spins: True
```
This and below are default values. I think you can remove them here.

abcdefg30 commented on mods/ra/rules/ships.yaml:
```diff
@@ -255,10 +228,7 @@ LST:
-	Explodes:
-		Weapon: UnitExplodeShip
-		EmptyWeapon: UnitExplodeShip
+	-Chronoshiftable:
```
Why that?

reaperrr replied:
*shrugs* The landing craft didn't have this trait before I moved it to the ^Ship default, either, so this is no change vs bleed. This was probably done to work around balancing issues or "kill load" issues, but I don't know.

reaperrr committed:
Fix Fort Lonestar and Monster Tank Madness

reaperrr commented:
Rebased.

reaperrr commented:
Rebased.

abcdefg30 commented on mods/ra/maps/monster-tank-madness/map.yaml:
```diff
@@ -2115,7 +2115,7 @@ Rules:
 			ShowOwnerRow: false
 		AnnounceOnSeen:
-	^Infantry:
+	^Soldier:
```
This doesn't look correct to me.
You're showing the `OwnerRow` of civilians etc now?

abcdefg30 commented:
Crashes when building a spy:
```
Red Alert Mod at Version {DEV_VERSION}
Operating System: Windows (Microsoft Windows NT 6.2.9200.0)
Runtime Version: .NET CLR 4.0.30319.34209
Exception of type `System.InvalidOperationException`: Actor does not have trait of type `OpenRA.Traits.IMove`
   at OpenRA.TraitDictionary.TraitContainer`1.Get(UInt32 actor) in \OpenRA\OpenRA.Game\TraitDictionary.cs:Line 148.
   at OpenRA.TraitDictionary.Get[T](Actor actor) in \OpenRA\OpenRA.Game\TraitDictionary.cs:Line 96.
   at OpenRA.Actor.Trait[T]() in \OpenRA\OpenRA.Game\Actor.cs:Line 212.
   at OpenRA.Mods.Common.Traits.AttackMove..ctor(Actor self, AttackMoveInfo info)
   at OpenRA.Mods.Common.Traits.AttackMoveInfo.Create(ActorInitializer init)
   at OpenRA.Actor..ctor(World world, String name, TypeDictionary initDict) in \OpenRA\OpenRA.Game\Actor.cs:Line 95.
   at OpenRA.World.CreateActor(Boolean addToWorld, String name, TypeDictionary initDict) in \OpenRA\OpenRA.Game\World.cs:Line 241.
   at OpenRA.World.CreateActor(String name, TypeDictionary initDict) in \OpenRA\OpenRA.Game\World.cs:Line 236.
   at OpenRA.Mods.Common.Traits.Production.<>c__DisplayClass6.<DoProduction>b__5(World w)
   at OpenRA.World.Tick() in \OpenRA\OpenRA.Game\World.cs:Line 319.
   at OpenRA.Game.InnerLogicTick(OrderManager orderManager) in \OpenRA\OpenRA.Game\Game.cs:Line 456.
   at OpenRA.Game.LogicTick() in \OpenRA\OpenRA.Game\Game.cs:Line 480.
   at OpenRA.Game.Loop() in \OpenRA\OpenRA.Game\Game.cs:Line 596.
   at OpenRA.Game.Run() in \OpenRA\OpenRA.Game\Game.cs:Line 636.
   at OpenRA.Program.Run(String[] args) in \OpenRA\OpenRA.Game\Support\Program.cs:Line 111.
   at OpenRA.Program.Main(String[] args) in \OpenRA\OpenRA.Game\Support\Program.cs:Line 39.

GraionDilach commented on mods/ra/rules/vehicles.yaml:
```diff
@@ -284,6 +265,9 @@ HARV:
 		Range: 4c0
+	-WithFacingSpriteBody:
+	-AutoSelectionSize:
+	-RenderSprites:
```
Why this?

penev92 replied:
Because of the line below: harvesters trade `RenderSprites` for (the soon to be deprecated (see #8553)) `RenderHarvester`.
@reaperrr might be a good idea to think about which of the PRs you want to go in first, so that there isn't any junk YAML left afterwards.

reaperrr replied:
@penev92: I'd like this to go in first, #8553 isn't yet 100% ready anyway.

reaperrr commented:
> Crashes when building a spy:

For the record, I managed to fix that by adding ```Mobile:``` to the SPY. Thing is, the spy inherits ```^Soldier``` like all other buildable infantry, meaning that crash makes absolutely no sense whatsoever. I suspect some whacky implicit dependency of one of the spy-exclusive traits is at work here, can't think of any other reason. That's out of this PR's scope, though.

GraionDilach commented:
https://github.com/OpenRA/OpenRA/blob/bleed/OpenRA.Mods.Common/Traits/Render/WithInfantryBody.cs#L20 must be the reason then.

penev92 commented:
According to the stack trace provided, it crashes at [this line](https://github.com/OpenRA/OpenRA/blob/bleed/OpenRA.Mods.Common/Traits/AttackMove.cs#L36), but I don't have the time to investigate now.

pchote commented:
AttackMoveInfo is missing a `Requires<IMoveInfo>`.

penev92 commented:
Ah, right. But that still doesn't explain the issue.

GraionDilach commented:
I still think WithInfantryBody removal triggers it somehow.

reaperrr commented:
Updated.

- fixed crashes related to spy and aircraft husks
- fixed tooltips on MTM
- made ant inherit ```^Infantry``` instead on ```^Soldier``` to disable weird idle animation

reaperrr commented:
Updated, the spy crash has been fixed in a separate PR.

GraionDilach commented:
Could you also remove https://github.com/OpenRA/OpenRA/blob/bleed/mods/ra/rules/misc.yaml#L156 with this? With that included, I'd say :+1:

reaperrr commented:
Updated.

abcdefg30 commented:
Looks good to me. :+1: / :white_check_mark: 

Could you squash the 3rd commit, please?

reaperrr commented:
Squashed.

penev92 commented:
Does this get a changelog entry?

Mailaender commented:
It doesn't really change anything for players or modders, just us.
